public/main.js

```diff
@@ -1,198 +1,155 @@
-var todoList = document.getElementById("todo-list");
-var todoListPlaceholder = document.getElementById("todo-list-placeholder");
-var form = document.getElementById("todo-form");
-var todoTitle = document.getElementById("new-todo");
-var error = document.getElementById("error");
-var completeCounter = document.getElementById("count-label");
-var deleteCompletedTodoBtn = document.getElementById("delete-complete-btn");
-var filterButtonAll = document.getElementById("filter-button-all");
-var filterButtonActive = document.getElementById("filter-button-active");
-var filterButtonComplete = document.getElementById("filter-button-completed");
-
-form.onsubmit = function(event) {
-    var title = todoTitle.value;
-    var dataToSend = JSON.stringify({
-        title: title
-    });
-    makeRequest("api/todo", "POST", dataToSend, "Failed to create item.")
-        .then(reloadTodoList);
-    todoTitle.value = "";
-    event.preventDefault();
-};
-
-deleteCompletedTodoBtn.onclick = function() {
-    makeRequest("api/todo/", "GET", null, "Failed to get list.")
-        .then(function(response) {
-            return response.json();
-        })
-        .then(function(todos) {
-            todos.forEach(function(todo) {
-                if (todo.isComplete) {
-                    makeRequest("api/todo/" + todo.id, "DELETE", null, "Failed to delete item.");
-                }
-            });
-        })
-        .then(reloadTodoList);
-};
-
-filterButtonAll.onclick = function() {
-    filterTodos("all");
-};
-filterButtonActive.onclick = function() {
-    filterTodos("active");
-};
-filterButtonComplete.onclick = function() {
-    filterTodos("complete");
-};
+/*global angular, _*/
+
+angular.module("todoList", ["ngAnimate"])
+    .controller("TodoListController", ["$scope", "$http", function($scope, $http) {
+
+        $scope.todos = [];
+        $scope.inputBoxText = "";
+        $scope.errorText = "";
+        $scope.todosLoaded = false;
+
+        $scope.getTodos = function() {
```

Comment:
As a stlying suggestion, I prefer declaring functions at the botton of the code and all scope delcarations separate at the top.
Have a look at https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md
Particularly the section called 'Bindable Members Up Top'
This comment applies to the rest of your code too.